test/buildingpart.test.js

```diff
@@ -0,0 +1,7 @@
+import { BuildingPart } from '../src/buildingpart.js';
+
+test('Test Cardinal to Degree', () => {
+  expect(BuildingPart.cardinalToDegree('N')).toBe(0);
+  expect(BuildingPart.cardinalToDegree('sSw')).toBe(202);
+  expect(BuildingPart.cardinalToDegree('Wse')).toBeUndefined();
+});
```

Comment:
Unit test is good; there are couple more cases that could be covered:
```suggestion
import { BuildingPart } from '../src/buildingpart.js';

test('Test Cardinal to Degree', () => {
  expect(BuildingPart.cardinalToDegree('N')).toBe(0);
  expect(BuildingPart.cardinalToDegree('sSw')).toBe(202);
  expect(BuildingPart.cardinalToDegree('Wse')).toBeUndefined();
  expect(BuildingPart.cardinalToDegree(0)).toBeUndefined();
  expect(BuildingPart.cardinalToDegree(null)).toBeUndefined();
  [['N', 0], ['NE', 45], ['E', 90], ['SE', 135], ['S', 180], ['SW', 225], ['W', 270], ['NW', 315], ['NNE', 22], ['ENE', 67], ['ESE', 112],
    ['SSE', 157], ['SSW', 202], ['WSW', 247], ['WNW', 292], ['NNW', 337]].forEach((a) => {
      expect(BuildingPart.cardinalToDegree(a[0])).toBe(a[1]);
    });
});
```

Tossed in an exhaustive test of expected values, plus a couple cases that should also be `undefined`.